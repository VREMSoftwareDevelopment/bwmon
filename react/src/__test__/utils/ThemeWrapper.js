import React from 'react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { create } from 'react-test-renderer';
import theme from './../../theme';

export const themeWrapper = (component) =>
    create(
        <StyledEngineProvider injectFirst>
            <ThemeProvider theme={theme}>{component}</ThemeProvider>
        </StyledEngineProvider>
    );
