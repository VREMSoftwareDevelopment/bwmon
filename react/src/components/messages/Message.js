/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { makeStyles } from '@mui/styles';
import { Alert, AlertTitle } from '@mui/material';

const useStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    },
}));

const Message = ({ severity, message }) => {
    const classes = useStyles();

    return message ? (
        <div className={classes.root}>
            <Alert severity={severity}>
                <AlertTitle>{message}</AlertTitle>
            </Alert>
        </div>
    ) : null;
};

export default Message;
