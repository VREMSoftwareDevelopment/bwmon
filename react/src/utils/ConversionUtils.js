/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import { DateTime } from 'luxon';

export const timeToDate = (time) => DateTime.fromSeconds(Number(time)).toFormat('MMM dd, yyyy HH:mm');
export const toMonth = (month) => DateTime.local(2012, month).toFormat('MMMM');
export const usageInGBytes = (value) => (Number(value) / 1000000).toFixed(3);
export const toPercent = (value) => Number(value).toFixed(1) + '%';
export const toIPv4 = (value) => (value >>> 24) + '.' + ((value >> 16) & 255) + '.' + ((value >> 8) & 255) + '.' + (value & 255);
export const fromIPv4 = (value) => value.split('.').reduce((value, octet) => (value << 8) + parseInt(octet, 10), 0) >>> 0;
