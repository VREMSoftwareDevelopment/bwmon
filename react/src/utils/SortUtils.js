/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

const isAscending = (orderBy, property, ascending) => (orderBy === property ? !ascending : false);

const comparator = (isAscending, orderBy) => {
    const ascending = (a, b, orderBy) => {
        if (a[orderBy] < b[orderBy]) return -1;
        if (a[orderBy] > b[orderBy]) return 1;
        return 0;
    };
    const descending = (a, b, orderBy) => {
        if (a[orderBy] > b[orderBy]) return -1;
        if (a[orderBy] < b[orderBy]) return 1;
        return 0;
    };
    return isAscending ? (a, b) => ascending(a, b, orderBy) : (a, b) => descending(a, b, orderBy);
};

const sort = (values, comparator) => [...values].sort(comparator);

export { comparator, isAscending, sort };
