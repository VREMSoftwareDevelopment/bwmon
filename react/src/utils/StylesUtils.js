import { makeStyles } from '@mui/styles';

const defaultStyles = makeStyles({});

const footerStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        margin: theme.spacing(2),
    },
}));

const graphStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        margin: theme.spacing(2),
        marginTop: 6,
    },
}));

const headerStyles = makeStyles({
    title: {
        flexGrow: 1,
    },
});

const messageStyles = makeStyles((theme) => ({
    root: {
        width: '100%',
        '& > * + *': {
            marginTop: theme.spacing(2),
        },
    },
}));

const paginationStyles = makeStyles((_theme) => ({
    root: {
        flexShrink: 0,
    },
}));

export { defaultStyles, footerStyles, graphStyles, headerStyles, messageStyles, paginationStyles };
