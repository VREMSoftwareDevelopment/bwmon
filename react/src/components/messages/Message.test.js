/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { themeWrapper } from '../../__test__/utils/ThemeWrapper';
import Message from './Message';

describe('Message', () => {
    it('renders correctly with message', () => {
        const tree = themeWrapper(<Message severity="success" message="message" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with no message', () => {
        const tree = themeWrapper(<Message severity="success" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
