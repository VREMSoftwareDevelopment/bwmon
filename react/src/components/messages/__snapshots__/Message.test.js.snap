// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly with message 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="SuccessOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        message
      </div>
    </div>
  </div>
</div>
`;

exports[`Message renders correctly with no message 1`] = `null`;
